tensorflow/contrib/memory_stats/python/kernel_tests/memory_stats_ops_test.py

```diff
@@ -64,10 +65,27 @@ def testChainOfMatmul(self):
       d = math_ops.matmul(c, b)
       sess.run(d)
 
-      max_bytes_in_use = sess.run(memory_stats_ops.MaxBytesInUse())
+      max_bytes_in_use_op = memory_stats_ops.MaxBytesInUse()
+      max_bytes_in_use = sess.run(max_bytes_in_use_op)
       self.assertGreaterEqual(max_bytes_in_use, matrix_size_in_bytes * 3)
       self.assertLess(max_bytes_in_use, matrix_size_in_bytes * 4)
 
+      # run chain with 2 ops, make sure BytesInUse captures intermediate
+      # memory usage
+      a = random_ops.random_uniform(matrix_shape, dtype=dtype)
+      with ops.control_dependencies([a]):
+        bytes_in_use_op = memory_stats_ops.BytesInUse()
+      with ops.control_dependencies([bytes_in_use_op]):
+        b = random_ops.random_uniform(matrix_shape, dtype=dtype)
+
+      _, bytes_in_use, max_bytes_in_use = sess.run([a, bytes_in_use_op,
+                                                    max_bytes_in_use_op])
+
+      # intermediate result allocates 1 matrix, max usage is at least 2
+      self.assertGreaterEqual(bytes_in_use, matrix_size_in_bytes * 1)
+      self.assertLess(bytes_in_use, matrix_size_in_bytes * 2)
+      self.assertGreaterEqual(max_bytes_in_use, matrix_size_in_bytes * 2)
```

Comment:
Can we make this more strict? max_bytes_in_use should >= matrix_size_in_bytes * 3, because as you said it's sticky to the allocator.

Reply:
I left it that way because I wasn't sure about guarantees about lifetime of the allocator. If the allocator gets reset between run calls this wouldn't happen. I'll assume allocator lifetime is guaranteed to span 2 session.run calls and update this in a bit